core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -284,13 +309,12 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
         transport = obtainActiveTransport();
       } catch (RuntimeException ex) {
         stream = new NoopClientStream();
-        listener.closed(Status.INTERNAL.withDescription("Failed starting transport").withCause(ex),
-            new Metadata.Trailers());
+        listener.closed(Status.fromThrowable(ex), new Metadata.Trailers());
```

Comment:
I'm expecting that the throwing code is going to throw a StatusRuntimeException. That's why this PR has the [OkHttp change](https://github.com/ejona86/grpc-java/commit/af8c63db5f42669ee2fe6db4f9c5eeec81326109#diff-464c935f90ab1b20df6aa2be2c8e1244R359) (which is also the only reason this code started handling the exception in 3462eb0).

If something really goes wrong and it isn't a StatusRuntimeException, then the full exception will still be present. Yes, in that case the status wouldn't be as rich, but I would also likely fix the code throwing the exception to throw a StatusRuntimeException. All that said, I don't really disagree with you, but having the description is a bit non-trivial for the various cases, and very little code will throw an exception here.

It's actually a "bug" that OkHttp's start() isn't asynchronous. If start() was always asynchronous, then obtainActiveTransport wouldn't really have an opportunity to throw an exception here.